### ISSUE TITLE
E3001 false positive: SAM resources accept (and silently drop) unknown resource-level properties

### ISSUE THREAD

**Issue by kddejong** [aws-cloudformation/cfn-lint#4606]
Reported by @paul-uz in https://github.com/aws-cloudformation/cfn-lint/pull/4491#issuecomment-5206595794

## Problem

cfn-lint 1.54.0 raises `E3001 Additional properties are not allowed` on `AWS::Serverless::Function` resources with properties at the resource level (sibling of `Type`/`Properties`), such as `Description`:

```yaml
Resources:
  MyFunction:
    Type: AWS::Serverless::Function
    Description: "At resource level"  # E3001 fires here
    Properties:
      Runtime: python3.12
      ...
```

However, this template **deploys successfully** via CloudFormation. The SAM transform silently drops unknown resource-level fields rather than rejecting them.

## Why this is a false positive

- **Deployment succeeds** — CloudFormation accepts the template
- **No runtime error** — the stack creates/updates without failure
- cfn-lint is flagging valid-to-deploy templates as errors

## Why this is still useful to flag

The SAM transform **silently drops** these fields. In the `Description` case, the Lambda function ends up with an empty description — the user's intent is lost without any indication. This applies to any unknown field (`Foo`, `TypoOfTimeout`, etc.) — all silently ignored.

Verified via real deployment:
- Template with `Description` at resource level → Lambda created with `Description: ""`
- Template with `Description` in `Properties` → Lambda has the description correctly

## Recommendation

Modify cfn-lint to:
1. **Not error** on unknown resource-level attributes when the resource type is `AWS::Serverless::*`
2. **Emit a warning** instead, indicating the field will be silently ignored by the SAM transform

SAM is unlikely to change this behavior (breaking backwards compatibility by rejecting previously-accepted templates), so cfn-lint should accommodate it.

## Workaround

Move `Description` (and any other properties) inside `Properties`:

```yaml
Resources:
  MyFunction:
    Type: AWS::Serverless::Function
    Properties:
      Description: "In Properties - works correctly"
      Runtime: python3.12
      ...
```

**Comment by paul-uz** [comment]
TIL the description should be in properties!

Thank you for looking into it though. I had to downgrade to 1.53.0 for now.
